### PR TITLE
bring ufs_utils to gaea

### PR DESCRIPTION
**Pull request by aerorahul**
This PR brings UFS utilities to Gaea.
HPC-stack is installed in my (@aerorahul) personal space without modules (LMod is not available on Gaea).
At some point, someone (other than me) will maintain the HPC-stack and a few lines can be replaced with appropriate`module load` commands.

@climbfuji wants to build the UFS on Gaea eventually using HPC-stack and this repo, since it uses (almost) everything that the UFS uses, was easy enough to port.
The HPC-stack issue installing on Gaea is https://github.com/NOAA-EMC/hpc-stack/issues/34

### CODE REVIEW

**Review by climbfuji — APPROVED** [review]
Should we record an issue to remind ourselves how the gaea build is configured?

I haven't heard from the gaea admins yet. If they push back, we may have to bite the bullet and convert the lua modulefiles to tcl. The latter works everywhere and provides the same functionality. I'd rather prefer not to, though.

**Comment by edwardhartnett** [comment]
Can you create an issue describing this and link to it?

**Comment by aerorahul** [comment]
> Should we record an issue to remind ourselves how the gaea build is configured?
> 
> I haven't heard from the gaea admins yet. If they push back, we may have to bite the bullet and convert the lua modulefiles to tcl. The latter works everywhere and provides the same functionality. I'd rather prefer not to, though.

@climbfuji 
There are several issues to consider:
- Tcl does not follow hierarchical module load and unload (to my knowledge).
- Writing Tcl modulefiles is easy, maintaining is going to be challenging and error prone.
- Build scripts may need some additions.

Most HPC's these days use LMod -- there are a few hold outs; Gaea and WCOSS Cray.  WCOSS Cray has an EOL set and we are moving away from that machine.  
Which machines do you support that don't have LMod? And can we get away with installing in a flat structure or a distributed flat structure for our applications.

Are there other alternatives and solutions?

**Comment by GeorgeGayno-NOAA** [comment]
> Can you create an issue describing this and link to it?

There is already a related issue - #152

**Review by GeorgeGayno-NOAA — COMMENTED** [review]
Thanks for doing this.  There is another PR ahead of you in the queue.  Will get to it soon.

**Comment by GeorgeGayno-NOAA** [comment]
@aerorahul If you merge the latest changes from 'develop' to your branch, I can close this today.

File: modulefiles/build.gaea.intel [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# hpc-stack installed as a flat install at:
+# /ncrc/home2/Rahul.Mahajan/dev/opt
+
+setenv ESMFMKFILE /ncrc/home2/Rahul.Mahajan/dev/opt/lib/esmf.mk
+
+prepend-path PATH /ncrc/home2/Rahul.Mahajan/dev/opt/bin
+prepend-path CMAKE_PREFIX_PATH /ncrc/home2/Rahul.Mahajan/dev/opt
+prepend-path LD_LIBRARY_PATH /ncrc/home2/Rahul.Mahajan/dev/opt/lib
+prepend-path LD_LIBRARY_PATH /ncrc/home2/Rahul.Mahajan/dev/opt/lib64
+
```
Comment (aerorahul) [inline modulefiles/build.gaea.intel:19]: FYI, I am not responsible for maintaining this space.
This is provided as a proof of concept, that UFS utils can indeed be built on Gaea.

**Comment by aerorahul** [comment]
> @aerorahul If you merge the latest changes from 'develop' to your branch, I can close this today.

@GeorgeGayno-NOAA 
done.

File: modulefiles/build.gaea.intel (hunk 1 above)
Comment (GeorgeGayno-NOAA) [reply modulefiles/build.gaea.intel:19]: Understood.

**Review by GeorgeGayno-NOAA — APPROVED** [review]
Changes add capability to compile on Gaea.  As a quick check, I was able to successfully build the branch (c4f3b95) on WCOSS-Dell, WCOSS-Cray, Hera and Orion.  Will merge.

File: modulefiles/build.gaea.intel (hunk 1 above)
Comment (climbfuji) [reply modulefiles/build.gaea.intel:19]: I can install hpc-stack on gaea in the project directories that we have been using for the ufs-weather-model support thus far, if needed.

**Pull request merged**